test/Conversion/LLHDToLLVM/convert_process.mlir

```diff
@@ -102,72 +148,332 @@ llhd.proc @convert_persistent_value () -> (%out0 : !llhd.sig<i1>, %out1 : !llhd.
+// CHECK:           %[[VAL_91:.*]] = llvm.extractvalue %[[VAL_72]][0 : i32] : !llvm.array<3 x i64>
+// CHECK:           %[[VAL_92:.*]] = llvm.extractvalue %[[VAL_72]][1 : i32] : !llvm.array<3 x i64>
+// CHECK:           %[[VAL_93:.*]] = llvm.extractvalue %[[VAL_72]][2 : i32] : !llvm.array<3 x i64>
+// CHECK:           %[[VAL_94:.*]] = llvm.call @llhdSuspend(%[[VAL_0]], %[[VAL_90]], %[[VAL_91]], %[[VAL_92]], %[[VAL_93]]) : (!llvm.ptr<i8>, !llvm.ptr<i8>, !llvm.i64, !llvm.i64, !llvm.i64) -> !llvm.void
+// CHECK:           llvm.return
+// CHECK:         ^bb10:
+// CHECK:           %[[VAL_95:.*]] = llvm.mlir.constant(4 : i32) : !llvm.i32
+// CHECK:           %[[VAL_96:.*]] = llvm.getelementptr %[[VAL_1]]{{\[}}%[[VAL_9]], %[[VAL_10]]] : (!llvm.ptr<struct<(ptr<i8>, i32, ptr<array<3 x i1>>, struct<(array<3 x i64>)>)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<i32>
+// CHECK:           llvm.store %[[VAL_95]], %[[VAL_96]] : !llvm.ptr<i32>
+// CHECK:           %[[VAL_97:.*]] = llvm.mlir.constant(0 : i32) : !llvm.i32
+// CHECK:           %[[VAL_98:.*]] = llvm.mlir.constant(2 : i32) : !llvm.i32
+// CHECK:           %[[VAL_99:.*]] = llvm.getelementptr %[[VAL_1]]{{\[}}%[[VAL_97]], %[[VAL_98]]] : (!llvm.ptr<struct<(ptr<i8>, i32, ptr<array<3 x i1>>, struct<(array<3 x i64>)>)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<ptr<array<3 x i1>>>
+// CHECK:           %[[VAL_100:.*]] = llvm.load %[[VAL_99]] : !llvm.ptr<ptr<array<3 x i1>>>
+// CHECK:           %[[VAL_101:.*]] = llvm.getelementptr %[[VAL_4]]{{\[}}%[[VAL_97]], %[[VAL_98]]] : (!llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<i64>
+// CHECK:           %[[VAL_102:.*]] = llvm.load %[[VAL_101]] : !llvm.ptr<i64>
+// CHECK:           %[[VAL_103:.*]] = llvm.mlir.constant(true) : !llvm.i1
+// CHECK:           %[[VAL_104:.*]] = llvm.getelementptr %[[VAL_100]]{{\[}}%[[VAL_97]], %[[VAL_102]]] : (!llvm.ptr<array<3 x i1>>, !llvm.i32, !llvm.i64) -> !llvm.ptr<i1>
+// CHECK:           llvm.store %[[VAL_103]], %[[VAL_104]] : !llvm.ptr<i1>
+// CHECK:           %[[VAL_105:.*]] = llvm.getelementptr %[[VAL_8]]{{\[}}%[[VAL_97]], %[[VAL_98]]] : (!llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<i64>
+// CHECK:           %[[VAL_106:.*]] = llvm.load %[[VAL_105]] : !llvm.ptr<i64>
+// CHECK:           %[[VAL_107:.*]] = llvm.mlir.constant(true) : !llvm.i1
+// CHECK:           %[[VAL_108:.*]] = llvm.getelementptr %[[VAL_100]]{{\[}}%[[VAL_97]], %[[VAL_106]]] : (!llvm.ptr<array<3 x i1>>, !llvm.i32, !llvm.i64) -> !llvm.ptr<i1>
+// CHECK:           llvm.store %[[VAL_107]], %[[VAL_108]] : !llvm.ptr<i1>
+// CHECK:           %[[VAL_109:.*]] = llvm.getelementptr %[[VAL_6]]{{\[}}%[[VAL_97]], %[[VAL_98]]] : (!llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<i64>
+// CHECK:           %[[VAL_110:.*]] = llvm.load %[[VAL_109]] : !llvm.ptr<i64>
```

Comment:
This seems more like an integration test than a unit test.  I'd recommend splitting this up so you test individual pieces.  This will make the test more stable and easier to maintain over time.

Reply:
@lattner I was wondering in general how we should handle integration tests. We have a few LLHD examples which we expect to produce a given simulation trace -- more like an end-to-end test case. Should we keep this in a separate repository, or somewhere here in CIRCT but away from the unit tests? I guess at some point we'll also want FIRRTL-to-Verilog or FIRRTL-to-Simulation-Trace tests that check end-to-end functionality.